src/Connector.test.js

```diff
@@ -0,0 +1,49 @@
+import Client, * as clientMocks from '@cesarbr/knot-cloud-sdk-js-amqp';
+import Connector from './Connector';
+
+jest.mock('@cesarbr/knot-cloud-sdk-js-amqp');
+
+const errors = {
+  connect: 'fail to connect to AMQP channel',
+};
+
+describe('Connector', () => {
+  beforeEach(() => {
+    clientMocks.mockConnect.mockClear();
+    clientMocks.mockRegister.mockClear();
+    clientMocks.mockUnregister.mockClear();
+    clientMocks.mockUpdateSchema.mockClear();
+    clientMocks.mockGetDevices.mockClear();
+    clientMocks.mockPublishData.mockClear();
+    clientMocks.mockOn.mockClear();
+    clientMocks.mockUnsubscribeEvent.mockClear();
+  });
+
+  test('start: should start connector when connection is stablished without errors', async () => {
+    const client = new Client();
+    const connector = new Connector(client);
+    await connector.start();
+    expect(clientMocks.mockConnect).toHaveBeenCalled();
+    expect(clientMocks.mockGetDevices).toHaveBeenCalled();
+  });
+
+  test('connect: should connect to client when there is no errors', async () => {
+    const client = new Client();
+    const connector = new Connector(client);
+    await connector.connectClient();
+    expect(clientMocks.mockConnect).toHaveBeenCalled();
+  });
+
+  test('connect: should fail to connect when something goes wrong', async () => {
+    const client = new Client({ connectErr: errors.connect });
+    const connector = new Connector(client);
+    let error;
+    try {
+      await connector.connectClient();
+    } catch (err) {
+      error = err.message;
+    }
+    expect(clientMocks.mockConnect).toHaveBeenCalled();
```

Comment:
It's strange that `toHaveBeenCalled()` is working here, since you need to spy a function to use this action. I did a research to see the examples on jest, and they use `spyOn` [here](https://jestjs.io/docs/en/jest-object#jestspyonobject-methodname).

Reply:
Good question! Note that `mockConnect` is a mock function created with `jest.fn()`. I'm using a mock implementation of AMQP SDK class (this one [here](https://jestjs.io/docs/en/es6-class-mocks#manual-mock)). The implementation is in `__mocks__/@cesarbr/knot-cloud-sdk-js-amqp`

The difference between mock with `jest.fn()` and `jest.spyOn()` is that the `spyOn` method maintain the function original implementation, allowing us to check if it's have been called, how many times it was called and another checks. The `fn` method also replaces the function implementation and allow us to define the values to be returned or in case of async methods, resolved/rejected.

Reply:
Oh, nice one @esdrasjnr! 

Just an informative commentary: using `spyOn`, you can define the values returned with `callFake(...)`, but since both `jest.fn()` and `jest.spyOn()` are able to use `toHaveBeenCalled()`, I don't see any problem to use it instead. Thank you for the explanation 💯

Reply:
Very insightful! :smile: